### PR TITLE
fix thread sync issue in pool

### PR DESCRIPTION
**Pull request by xrdavies**
fix thread sync issue in pool

### CODE REVIEW

File: client/pool.c [hunk 1]
```diff
@@ -1092,11 +1110,27 @@ int xdag_print_miners(FILE *out)
...
-	LL_FOREACH(g_miner_list_head, elt)
-	{
-		struct miner_pool_data *miner = &elt->miner_data;
-		count_active += print_miner(out, index++, miner, 1);
-	}
+	
+	//if not locked, wait for 100 ms, max try time 10.
+	int times = 0;
+	do {
+		if(!pthread_mutex_trylock(&g_miners_mutex)) {
```
Comment (jonano614) [inline client/pool.c]: Am I right deadlock are possible here?

File: client/miner.c [hunk 2]
```diff
@@ -76,7 +77,6 @@ static int send_to_pool(struct xdag_field *fld, int nfld)
 	int i, res, todo = nfld * sizeof(struct xdag_field), done = 0;
 
 	if(g_socket < 0) {
-		pthread_mutex_unlock(&g_pool_mutex);
```
[the remark this answers is not in the document]
Comment (xrdavies) [reply client/miner.c]: Removed to outside.

File: client/pool.c (hunk 1 above)
Comment (xrdavies) [reply client/pool.c]: Yes, possible. Please find updates.

File: client/miner.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonano614) [reply client/miner.c]: ok

**Pull request merged**